registry/client/src/apps/Edit.js

```diff
@@ -1,74 +1,170 @@
-import React from 'react';
+import React, {Fragment} from 'react';
+
 import {
     Create,
     Edit,
     FormTab,
     SelectInput,
     TabbedForm,
     TextInput,
-    NumberInput,
-    required,
     ArrayInput,
     SimpleFormIterator,
+    FormDataConsumer,
+    NumberInput,
     TextField,
     ReferenceArrayInput,
-    AutocompleteArrayInput
+    AutocompleteArrayInput,
 } from 'react-admin'; // eslint-disable-line import/no-unresolved
+
 import JsonField from '../JsonField/index';
+import * as validators from '../validators';
 
-const Title = ({ record }) => {
-    return (<span>{record ? `App "${record.name}"` : ''}</span>);
+const selectHasAssetsDiscoveryUrl = (formData) => formData.assetsDiscoveryUrl && formData.assetsDiscoveryUrl.length !== 0;
+const selectWarnMessageDueToAssetsDiscoveryUrl = (formData) => {
+    if (!selectHasAssetsDiscoveryUrl(formData)) {
+        return '';
+    }
+
+    return `Do not need to specify SPA bundle, CSS bundle, dependencies because they would be fetched and set from assets discovery URL if they exist there`;
+};
+
+const validateAppCreation = (values) => {
```

Comment:
validateApp

Reply:
The code below is really complex. I don't really like the approach....

Reply:
Renamed
Removed unnecessary code but I left necessary validation here for 'assetsDiscoveryUrl', 'spaBundle'